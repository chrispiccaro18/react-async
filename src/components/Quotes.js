import React from 'react';
import PropTypes from 'prop-types';
import Quote from './Quote';

function Quotes({ quotes }) {
  const quoteLis = quotes.map(quoteObj => {
    const { quote, character, image } = quoteObj;
    return <Quote key={`${quote}${character}${image}`} quote={quote} character={character} image={image} />;
  });

  return (
    <ul>
      {quoteLis}
    </ul>
  );
}

Quotes.propTypes = {
  quotes: PropTypes.arrayOf(
    PropTypes.shape({
      quote: PropTypes.string.isRequired,
      character: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired
    })
  )
};

export default Quotes;
