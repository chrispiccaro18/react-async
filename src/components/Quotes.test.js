import React from 'react';
import { shallow } from 'enzyme';
import Quotes from './Quotes';

describe('Quotes component', () => {
  it('renders Quotes component', () => {
    const quotes = [
      {
        'character': 'Bender',
        'quote': 'Problem solved. You two fight to the death and I\'ll cook the loser.',
        'image': 'https://res.cloudinary.com/dzxqhkyqd/image/fetch/c_scale,w_500/https://res.cloudinary.com/dzxqhkyqd/image/upload/v1552429540/bender.png'
      },
      {
        'character': 'Bender',
        'quote': 'Bite my shiny metal ass.',
        'image': 'https://res.cloudinary.com/dzxqhkyqd/image/fetch/c_scale,w_500/https://res.cloudinary.com/dzxqhkyqd/image/upload/v1552429540/bender.png'
      },
      {
        'character': 'Professor-farnsworth',
        'quote': 'I\'m sure Bender has just made some cunning remark, but he doesn\'t know I taped\nover his soap operas to record this message.',
        'image': 'https://res.cloudinary.com/dzxqhkyqd/image/upload/v1554904014/farnsworth.png'
      },
      {
        'character': 'Bender',
        'quote': 'You\'re a thousand years old?! Your bio says you\'re 27!',
        'image': 'https://res.cloudinary.com/dzxqhkyqd/image/fetch/c_scale,w_500/https://res.cloudinary.com/dzxqhkyqd/image/upload/v1552429540/bender.png'
      },
      {
        'character': 'Zapp-brannigan',
        'quote': 'You\'ll be negotiating with the mysterious brain-balls. They\'ve got a lot of\nbrains and a lot of... chutzpah.',
        'image': 'https://res.cloudinary.com/dzxqhkyqd/image/upload/v1554904049/Zapp_Brannigan.png'
      }
    ];
    
    const wrapper = shallow(<Quotes quotes={quotes} />);
    expect(wrapper).toMatchSnapshot();
  });
});
