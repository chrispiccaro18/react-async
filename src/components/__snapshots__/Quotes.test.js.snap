// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes component renders Quotes component 1`] = `
<ul>
  <Quote
    character="Bender"
    image="https://res.cloudinary.com/dzxqhkyqd/image/fetch/c_scale,w_500/https://res.cloudinary.com/dzxqhkyqd/image/upload/v1552429540/bender.png"
    key="Problem solved. You two fight to the death and I'll cook the loser.Benderhttps://res.cloudinary.com/dzxqhkyqd/image/fetch/c_scale,w_500/https://res.cloudinary.com/dzxqhkyqd/image/upload/v1552429540/bender.png"
    quote="Problem solved. You two fight to the death and I'll cook the loser."
  />
  <Quote
    character="Bender"
    image="https://res.cloudinary.com/dzxqhkyqd/image/fetch/c_scale,w_500/https://res.cloudinary.com/dzxqhkyqd/image/upload/v1552429540/bender.png"
    key="Bite my shiny metal ass.Benderhttps://res.cloudinary.com/dzxqhkyqd/image/fetch/c_scale,w_500/https://res.cloudinary.com/dzxqhkyqd/image/upload/v1552429540/bender.png"
    quote="Bite my shiny metal ass."
  />
  <Quote
    character="Professor-farnsworth"
    image="https://res.cloudinary.com/dzxqhkyqd/image/upload/v1554904014/farnsworth.png"
    key="I'm sure Bender has just made some cunning remark, but he doesn't know I taped
over his soap operas to record this message.Professor-farnsworthhttps://res.cloudinary.com/dzxqhkyqd/image/upload/v1554904014/farnsworth.png"
    quote="I'm sure Bender has just made some cunning remark, but he doesn't know I taped
over his soap operas to record this message."
  />
  <Quote
    character="Bender"
    image="https://res.cloudinary.com/dzxqhkyqd/image/fetch/c_scale,w_500/https://res.cloudinary.com/dzxqhkyqd/image/upload/v1552429540/bender.png"
    key="You're a thousand years old?! Your bio says you're 27!Benderhttps://res.cloudinary.com/dzxqhkyqd/image/fetch/c_scale,w_500/https://res.cloudinary.com/dzxqhkyqd/image/upload/v1552429540/bender.png"
    quote="You're a thousand years old?! Your bio says you're 27!"
  />
  <Quote
    character="Zapp-brannigan"
    image="https://res.cloudinary.com/dzxqhkyqd/image/upload/v1554904049/Zapp_Brannigan.png"
    key="You'll be negotiating with the mysterious brain-balls. They've got a lot of
brains and a lot of... chutzpah.Zapp-branniganhttps://res.cloudinary.com/dzxqhkyqd/image/upload/v1554904049/Zapp_Brannigan.png"
    quote="You'll be negotiating with the mysterious brain-balls. They've got a lot of
brains and a lot of... chutzpah."
  />
</ul>
`;
